### PR TITLE
Issue #SB-17576 feat: added multi select functionality

### PR DESCRIPTION
**Pull request by Diptesh6501**

### CODE REVIEW

File: projects/quml-library/src/lib/mcq-option/mcq-option.component.ts [hunk 1]
```diff
@@ -10,8 +10,10 @@ export class McqOptionComponent implements OnInit {
   @Input() mcqOptions: any;
   @Input() solutions: any;
   @Input() layout: any;
+  @Input() cardinality: any;
```
Comment (HarishGangula) [inline projects/quml-library/src/lib/mcq-option/mcq-option.component.ts:13]: It is not any set the type

File: projects/quml-library/src/lib/player/player.component.ts [hunk 2]
```diff
@@ -231,29 +231,54 @@ export class PlayerComponent implements OnInit, AfterViewInit {
...
-        this.showAlert = true;
-        this.alertType = false;
-        if (this.showFeedBack) {
-          this.updateScoreBoard((currentIndex + 1), 'wrong');
-        }
-        if (!this.showFeedBack) {
-          this.nextSlide();
+      }
+      if (option.cardinality === 'multiple') {
+        let key:any  = await this.utilService.getKeyValue(Object.keys(this.questions[currentIndex].responseDeclaration));
```
Comment (HarishGangula) [inline projects/quml-library/src/lib/player/player.component.ts:262]: Here correct Response is not checked

File: projects/quml-library/src/lib/util-service.ts [hunk 3]
```diff
@@ -35,4 +35,21 @@ export class UtilService {
         })
     }
 
+    public getMultiselectScore(options, mappings) {
+        return new Promise((resolve, reject) => {
```
Comment (HarishGangula) [inline projects/quml-library/src/lib/util-service.ts:39]: Why promise is required for synchronous operation, it is unnnessary

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: projects/quml-library/src/lib/player/player.component.ts (hunk 2 above)
Comment (Diptesh6501) [reply projects/quml-library/src/lib/player/player.component.ts:262]: i did not understand , can you mention it more clearly

File: projects/quml-library/src/lib/util-service.ts (hunk 3 above)
Comment (Diptesh6501) [reply projects/quml-library/src/lib/util-service.ts:39]: this is required because without this the score is coming as undefined , it has been tested

**Review by HarishGangula — CHANGES_REQUESTED** [review]
As discussed fix all the issue and update PR

File: projects/quml-library/src/lib/startpage/startpage.component.html [hunk 4]
```diff
@@ -10,10 +10,7 @@
       General Instructions
     </div>
     <div class="startpage-guidelines">
-      <p>Read and understand the test guidelines</p>
-      <p>Click on audio icon to play sound</p>
-      <p>Tap on the zoom icon to enlarge the image</p>
-      <p>Tap on the option to select answer</p>
+      {{instructions}}
```
Comment (HarishGangula) [inline projects/quml-library/src/lib/startpage/startpage.component.html:13]: instructions can have HTML tags as well

File: projects/quml-library/src/lib/util-service.ts [hunk 5]
```diff
@@ -26,13 +27,34 @@ export class UtilService {
...
+        let key = keys.find((k) => {
+            return k.includes('response');
         })
+        return key;
+    }
+
+    public getMultiselectScore(options, responseDeclaration) {
+        let key: any = this.getKeyValue(Object.keys(responseDeclaration));
+        const selectedOptionValue = options.map(option => option.value);
+        let score = responseDeclaration[key].correct_response.outcomes.score ? responseDeclaration[key].correct_response.outcomes.score : responseDeclaration.maxScore;
```
Comment (HarishGangula) [inline projects/quml-library/src/lib/util-service.ts:39]: check key **correct_response** from meta data and fix

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: projects/quml-library/src/lib/service/user-service/user-service.ts [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+import { QumlPlayerConfig } from '../../quml-library-interface';
+import { QumlLibraryService } from '../../quml-library.service';
+import { UtilService } from '../../util-service';
+import { eventName, TelemetryType } from '../../telemetry-constants';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
```
Comment (HarishGangula) [inline projects/quml-library/src/lib/service/user-service/user-service.ts:11]: Why this is called user service ?

File: projects/quml-library/src/lib/player/player.component.ts [hunk 7]
```diff
@@ -34,7 +34,7 @@ export class PlayerComponent implements OnInit, AfterViewInit {
   currentSolutions: any;
   showSolution: any;
   active = false;
-  alertType: boolean;
+  alertType: string;
```
Comment (HarishGangula) [inline projects/quml-library/src/lib/player/player.component.ts:37]: Need test cases for this file

**Review by HarishGangula — CHANGES_REQUESTED** [review]

**Review by HarishGangula — APPROVED** [review]

**Pull request merged**